framework/tlvf/yaml/tlvf/wfa_map/tlvApMetric.yaml

```diff
@@ -9,22 +9,22 @@ tlvApMetric:
     _type: eTlvTypeMap
     _value_const: TLV_AP_METRIC 
   length: uint16_t
-  bssid: sAddress
+  bssid: sMacAddr
   channel_utilization: uint8_t
   number_of_stas_currently_associated: uint16_t
   estimated_service_parameters: sEstimatedService
   estimated_service_info_field_ac_be:
     _type: uint8_t
-    _length: [ ]
+    _length: [3]
```

Comment:
As a permanent solution for the problem of the optional fields, you could define just this one field and make it dynamic:

```yaml
  estimated_service_info:
    _type: uint8_t
    _length: [ ]
```

The receiver can validate if the length corresponds to what is expected based on the estimated_service_parameters.

Reply:
the problem that it doesn't pass compilation.

Reply:
> the problem that it doesn't pass compilation.

It builds for me...

Reply:
It does not compile for me also @arnout, maybe you meant  to define a length variable as defined here :
`
cRestrictedOperatingClasses:
  _type: class
  operating_class: uint8_t
  channel_list_length:
    _type: uint8_t
    _length_var: True
  channel_list:
    _type: sChannelInfo
    _length: [ channel_list_length ]
`

Reply:
> It does not compile for me also @arnout, maybe you meant to define a length variable as defined here?

There is no length variable that can be used in the AP_METRICS structure, only the overall length of the TLV.

I did try it with exactly the code I wrote:
```
  estimated_service_info:
    _type: uint8_t
    _length: [ ]
```

and it built successfully (after, of course, adapting all the uses of estimated_service_info_*).

Reply:
@CoralMalachi , @kantera800 you both didn't understand @arnout 's suggestion.
Instead of having 4 different unknown length lists which are optional based on the `estimated_service_parameters`, he suggested to replace it with a single unknown length list.
The failure you originally saw was that TLVF cannot accommodate more then one unknown length list OR an unknown length list which is not the *last* member in the TLV.
The reason for that is due to the fact that on parsing, we use the TLV `length` field and subtract all the rest of the class parameter sizes from it to get the unknown length list actual length.

So - instead of having 4 different unknown length lists which by now you should understand why it is not supported, lets define a single one:

```yaml
tlvApMetric:
  _type: class
  _is_tlv_class : True
  type:
    _type: eTlvTypeMap
    _value_const: TLV_AP_METRIC 
  length: uint16_t
  bssid: sMacAddr
  channel_utilization: uint8_t
  number_of_stas_currently_associated: uint16_t
  estimated_service_parameters: sEstimatedService
  estimated_service_info:
    _type: uint8_t
    _length: []
```

Usage:
First, understand the length of the info based on `sEstimatedService`, lets say it is `len`.
Then we simply allocate this list:
```
ap_metrics->alloc_estimated_service_info(len);
auto info = ap_metrics->estimated_service_info(); //get the buffer
```
Now we can access the buffer and modify it as we need based on `sEstimatedService`.

However - this PR is blocked for too long and I really want to get it merged, and there is already a task for this - #473 - so let's keep the WA as is since by miracle it passes the test and @CoralMalachi will fix very soon as part of a separate PR.